packages/near-swap-provider/lib/NearSwapProvider.ts

```diff
@@ -0,0 +1,157 @@
+import { SwapProvider, SwapParams, near } from "@liquality/types";
+
+import Provider from "@liquality/provider";
+import { PendingTxError, TxNotFoundError } from "@liquality/errors";
+import { toNearTimestampFormat, parseReceipt } from "@liquality/near-utils";
+import BigNumber from "bignumber.js";
+import { transactions } from "near-api-js";
+
+import Bytecode from "./bytecode";
+
+const CONTRACT_CODE = "wnGY4a+YYkfbkBqYqyhdQjjtHRl9Auyh7yxCGtXL8JI=";
+
+const ABI = {
+  init: { method: "init", gas: "10000000000000" },
+  claim: { method: "claim", gas: "10000000000000" },
+  refund: { method: "refund", gas: "8000000000000" },
+};
+
+export default class NearSwapProvider
+  extends Provider
+  implements Partial<SwapProvider> {
+  createSwapScript() {
+    return Bytecode;
+  }
+
+  async initiateSwap(
+    swapParams: SwapParams
+  ): Promise<near.NormalizedTransaction> {
+    const bytecode = this.createSwapScript();
+    const contractId = this.generateUniqueString();
+
+    return this.getMethod("sendTransaction")(contractId, null, [
+      transactions.createAccount(),
```

Comment:
It seems like you are going to have an issue here... in that you wanted to pass more params to `sendTransaction` but the interface for sendtransaction does not allow this. 

I think either you can modify `SendOptions` so that `data` is any type. This way, you can pass any object as the data part to be processed, in this case your actions